compio-driver/src/poll/mod.rs

```diff
@@ -352,32 +391,22 @@ impl Driver {
                         .get_mut(&fd)
                         .expect("the fd should be attached");
                     if let Some((user_data, interest)) = queue.pop_interest(&event) {
-                        if self.cancelled.remove(&user_data) {
-                            entry_cancelled(user_data).notify();
-                        } else {
-                            let mut op = Key::<dyn crate::sys::OpCode>::new_unchecked(user_data);
-                            let op = op.as_op_pin();
-                            let res = match op.operate() {
-                                Poll::Pending => {
-                                    // The operation should go back to the front.
-                                    queue.push_front_interest(user_data, interest);
-                                    None
-                                }
-                                Poll::Ready(res) => Some(res),
-                            };
-                            if let Some(res) = res {
-                                Entry::new(user_data, res).notify();
+                        let mut op = Key::<dyn crate::sys::OpCode>::new_unchecked(user_data);
+                        let op = op.as_op_pin();
+                        let res = match op.operate() {
+                            Poll::Pending => {
+                                // The operation should go back to the front.
+                                queue.push_front_interest(user_data, interest);
+                                None
                             }
+                            Poll::Ready(res) => Some(res),
+                        };
+                        if let Some(res) = res {
+                            Entry::new(user_data, res).notify();
                         }
                     }
                     let renew_event = queue.event();
-                    let borrowed_fd = BorrowedFd::borrow_raw(fd);
-                    if !renew_event.readable && !renew_event.writable {
-                        self.poll.delete(borrowed_fd)?;
-                        self.registry.remove(&fd);
-                    } else {
-                        self.poll.modify(borrowed_fd, renew_event)?;
-                    }
+                    Self::renew(&self.poll, &mut self.registry, fd, renew_event)?;
```

Comment:
Why is renewing necessary? If it's cancelled no future events should be expected?